src/overlays/actors/ovl_En_Sekihi/z_en_sekihi.c

```diff
@@ -32,20 +36,147 @@ const ActorInit En_Sekihi_InitVars = {
+
+    params = ENSIKIHI_GET_F(&this->dyna.actor);
+    if (Object_IsLoaded(&play->objectCtx, this->objectIndex)) {
+        this->dyna.actor.objBankIndex = this->objectIndex;
+        this->dyna.actor.draw = EnSekihi_Draw;
+        if (params == 4) {
+            this->actionFunc = func_80A450B0;
+        } else {
+            this->actionFunc = func_80A45130;
+        }
+
+        Actor_SetObjectDependency(play, &this->dyna.actor);
+        DynaPolyActor_Init(&this->dyna, 0);
+        if (sColHeader[params] != 0) {
+            CollisionHeader_GetVirtual(sColHeader[params], &colHeader);
+        }
+
+        this->dyna.bgId = DynaPoly_SetBgActor(play, &play->colCtx.dyna, &this->dyna.actor, colHeader);
+        if ((params == 4) && (INV_CONTENT(ITEM_MASK_ZORA) != ITEM_MASK_ZORA)) {
+            Actor_MarkForDeath(&this->dyna.actor);
+        }
+    }
+}
+
+void func_80A44F40(EnSekihi* this, PlayState* play) {
+    switch (Message_GetState(&play->msgCtx)) {
+        case 4:
+            if ((Message_ShouldAdvance(play)) && (play->msgCtx.currentTextId == 0x1019)) {
+                switch (play->msgCtx.choiceIndex) {
+                    case 0:
+                        func_8019F208();
+                        func_80151938(play, 0x101A);
+                        break;
+                    case 1:
+                        func_8019F208();
+                        func_80151938(play, 0x101B);
+                        break;
+                    case 2:
+                        func_8019F230();
+                        func_801477B4(play);
+                        this->actionFunc = func_80A450B0;
+                        break;
+                }
+            } else {
+                break;
+            }
+            break;
+        case 5:
+            if (Message_ShouldAdvance(play)) {
+                switch (play->msgCtx.currentTextId) {
+                    case 0x1018:
+                        func_80151938(play, (play->msgCtx.currentTextId + 1) & 0xFFFF);
+                        break;
+                    case 0x101A:
+                    case 0x101B:
+                        func_801477B4(play);
+                        this->actionFunc = func_80A450B0;
+                        break;
+                }
+            }
+            break;
+    }
+}
+
+void func_80A450B0(EnSekihi* this, PlayState* play) {
+    if (Actor_ProcessTalkRequest(&this->dyna.actor, &play->state)) {
+        this->actionFunc = func_80A44F40;
+    } else if ((this->dyna.actor.xzDistToPlayer < 100.0f) && (Player_IsFacingActor(&this->dyna.actor, 0x2600, play))) {
+        func_800B8614(&this->dyna.actor, play, 120.0f);
+    }
+}
+
+void func_80A45130(EnSekihi* this, PlayState* play) {
+}
+
+void EnSekihi_Update(Actor* thisx, PlayState* play) {
+    EnSekihi* this = THIS;
+
+    this->actionFunc(this, play);
+}
+
+void EnSekihi_Draw(Actor* thisx, PlayState* play) {
+    EnSekihi* this = THIS;
+
+    OPEN_DISPS(play->state.gfxCtx);
+
+    if (this->xluDList != 0) {
```

Comment:
```suggestion
    if (this->xluDList != NULL) {
```

Reply:
Errors out as this->xluDList is an s32

Reply:
Both `xluDList` and `opaDList` shouldn't be `s32`. They should be `Gfx*`.